design-proposals/project-deploymnet.md

```diff
@@ -0,0 +1,176 @@
+# Design Proposal: Project Deployment
+
+**Authors**: @kfirtoledo 
+
+**Begin Design Discussion**: 2023-11-30
+
+**Status:** draft
+
+## Summary/Abstract
+
+This design proposal outlines the deployment process for the ClusterLink project on a Kubernetes (K8s) cluster.  
+The proposal specifically addresses deployment within the K8s environment.   
+It will rely on a dedicated K8s operator, to make the deployment more efficient, user-friendly, and native to K8s users.
+
+## Background
+
+### Motivation and problem space
+
+This design proposal aims to simplify and enhance the deployment of the Clusterlink project.
+By leveraging K8s deployment standards, such as the K8s operator, the objective is to establish a user-friendly deployment process.
+
+### Impact and desired outcome
+
+Implementing this proposal is expected to improve the accessibility and ease of use for Clusterlink users.   
+Aligning the deployment process with K8s standards will particularly benefit use cases involving central control management, such as Multi-Cloud Computing (MCC) or Hybrid Cloud Mesh (HCM).
+
+### Prior discussion and links
+
+Not applicable.
+
+## User/User Story
+
+This is an example to user guide with all the details to install the ClusterLink project on a K8s Cluster.
+
+### Get ClusterLink project
+Clone the ClusterLink project:
+
+    git clone git@github.com:clusterlink-net/clusterlink.git
+
+Install the ClusterLink CLI:
+
+    cd clusterlink
+    make install
+
+In the future, these two commands should be replaced the command:
+
+    curl -L https://clusterlink.net/install.sh | sh
+
+### Create Fabric certificate
+Create the fabric certificate, which is the root CA certificate for all the K8s clusters:
+
+    clusterlink create fabric --name <fabric_name>
```

Comment:
I see the following commands run by a *fabric administrator*:
- `clusterlink create fabric <fabric_name>` - produces CA certificate and key 
- `clusterlink create site --fabric <fabric_name> --name <site_name>` - produces site certificate and key

then the ca.crt, <site_name>.crt and <site_name>.key can be distributed to the clusterlink admin of a specific cluster.
The two steps can be automated in a central management environment.